compliance_tool/cli.py

```diff
@@ -1,57 +1,67 @@
 import argparse
-from compliance_tool import list, scan, remediate
+from compliance_tool import list_, scan, remediate
 
 
-def prepare_parser_list(subparsers) -> None:
-    parser = subparsers.add_parser("list",
-                                   help="list available components")
-    parser.set_defaults(func=list.execute)
+def prepare_parser_list(cli_sub_parsers, global_parser, profile_parser) -> None:
```

Comment:
I'm currently getting quite awful user experience with this parser.

```
jcerny@fedora:~/work/git/compliance-tool (pr/10)$ python3 -m compliance_tool.cli list
usage: cli.py list [-h] [--input SOURCE_FILE] [--tailoring SOURCE_FILE] {profiles,controls,rules} ...
cli.py list: error: the following arguments are required: {profiles,controls,rules}
```

We strongly need to improve this in order to provide a good user experience.

Reply:
So, I've made this sub-parser optional. By default it'll list profiles (but if `profiles` sub-command is given it won't object as well).
```
compliance-tool list                                      # list proflies, implicit profiles
compliance-tool list profiles                        # list profiles, no warning, just for consistency
compliance-tool list rules --profile=blah   # list rules, --profile is mandatory
```